src/plugins/application_config/server/opensearch_config_client.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { IScopedClusterClient, Logger } from '../../../../src/core/server';
+
+import { ConfigurationClient } from './types';
+
+export class OpenSearchConfigurationClient implements ConfigurationClient {
+  private client: IScopedClusterClient;
+  private configurationIndexName: string;
+  private readonly logger: Logger;
+
+  constructor(
+    openSearchClient: IScopedClusterClient,
+    configurationIndexName: string,
+    logger: Logger
+  ) {
+    this.client = openSearchClient;
+    this.configurationIndexName = configurationIndexName;
+    this.logger = logger;
+  }
+
+  async getEntityConfig(entity: string) {
```

Comment:
_This doesn't need to in the scope of this PR._

Can we add validation for entity to not allow bad entity string?

Reply:
added input validation as well as related unit tests.